cmd/clusterctl/pkg/client/cluster/template.go

```diff
@@ -0,0 +1,207 @@
+	}
+
+	return repository.NewTemplate([]byte(data), t.configClient.Variables(), targetNamespace, listVariablesOnly)
+}
+
+func (t *templateClient) GetFromURL(templateURL, targetNamespace string, listVariablesOnly bool) (repository.Template, error) {
+	if templateURL == "" {
+		return nil, errors.New("invalid GetFromURL operation: missing templateURL value")
+	}
+
+	content, err := t.getURLContent(templateURL)
+	if err != nil {
+		return nil, err
+	}
+
+	return repository.NewTemplate(content, t.configClient.Variables(), targetNamespace, listVariablesOnly)
+}
+
+func (t *templateClient) getURLContent(templateURL string) ([]byte, error) {
+	rURL, err := url.Parse(templateURL)
+	if err != nil {
+		return nil, errors.Wrapf(err, "invalid GetFromURL operation: failed to parse %q", templateURL)
+	}
+
+	if rURL.Scheme == "https" && rURL.Host == "github.com" {
+		return t.getGitHubFileContent(rURL)
+	}
+
+	if rURL.Scheme == "File" || rURL.Scheme == "" {
+		return t.getLocalFileContent(rURL)
+	}
+
+	return nil, errors.Errorf("unable to read content from %q. Only reading from GitHub and local file system is supported", templateURL)
+}
+
+func (t *templateClient) getLocalFileContent(rURL *url.URL) ([]byte, error) {
+	f, err := os.Stat(rURL.Path)
+	if err != nil {
+		return nil, errors.Errorf("failed to read file %q", rURL.Path)
+	}
+	if f.IsDir() {
+		return nil, errors.Errorf("invalid path: file %q is actually a directory", rURL.Path)
+	}
+	content, err := ioutil.ReadFile(rURL.Path)
+	if err != nil {
+		return nil, errors.Wrapf(err, "failed to read file %q", rURL.Path)
+	}
+
+	return content, nil
+}
+
+func (t *templateClient) getGitHubFileContent(rURL *url.URL) ([]byte, error) {
+	// Check if the path is in the expected format,
+	urlSplit := strings.Split(strings.TrimPrefix(rURL.Path, "/"), "/")
+	if len(urlSplit) < 5 {
+		return nil, errors.Errorf(
+			"invalid GitHub url %q: a GitHub url should be in the form https://github.com/{owner}/{repository}/blob/{branch}/{path-to-file}", rURL,
+		)
+	}
+
+	// Extract all the info from url split.
+	owner := urlSplit[0]
+	repository := urlSplit[1]
+	branch := urlSplit[3]
+	path := strings.Join(urlSplit[4:], "/")
+
+	// gets the GitHub client
+	client, err := t.gitHubClientFactory(t.configClient.Variables())
+	if err != nil {
+		return nil, err
+	}
+
+	// gets the file from GiHub
+	fileContent, _, _, err := client.Repositories.GetContents(context.TODO(), owner, repository, path, &github.RepositoryContentGetOptions{Ref: branch})
+	if err != nil {
+		return nil, handleGithubErr(err, "failed to get %q", rURL.Path)
+	}
+
+	if fileContent == nil {
+		return nil, errors.Errorf("%q does not returns a valid file content", rURL.Path)
+	}
+
+	if fileContent.Encoding == nil || *fileContent.Encoding != "base64" {
+		return nil, errors.Errorf("invalid encoding detected for %q. Only base 64 encoding supported", rURL.Path)
```

Comment:
nit:
```suggestion
		return nil, errors.Errorf("invalid encoding detected for %q. Only base64 encoding supported", rURL.Path)
```

Reply:
Also wanted to point out that we are just printing the `rURL.Path` and not `rURL.String()`. Not sure if that was the behavior you wanted. 🙂

Reply:
I found printing rURL.Path less verbose, but I don't have strong opinions

Reply:
Cool...I'm fine with it. Just wanted to confirm expected behavior.